### PR TITLE
VM versions as HashMap

### LINKED ISSUE #70: Allow to fetch all VMs versions from `info.getNodeVersion`
Currently, the library uses the `VmVersions` struct to deserialize the `vmVersions` field of [info.getNodeVersion](https://docs.avax.network/apis/avalanchego/apis/info#infogetnodeversion). This struct only supports the 3 VMs of the Primary Network:
https://github.com/ava-labs/avalanche-types-rs/blob/6c7603418b9b00d5785c9de4ecf8f93fab6a3542/src/jsonrpc/info.rs#L305-L309

As the number of Subnets/VMs increase, we should be able to parse any number of VMs as a `HashMap`.

### PR DESCRIPTION
**Pull request by Nuttymoon**
Fixes #70 

### Changes
- Add a HashMap to store the Subnet's VM versions in `VmVersions`
- Add structs to query `platform.getPendingValidators` + associated test
- Made the comments on `ApiPrimaryValidator` and `ApiPrimaryDelegator` fields more explicit ("elastic/permissioned" Subnet)

### Additional comments
- I reused the existing `ApiPrimaryValidator` and `ApiPrimaryDelegator` for `GetPendingValidators`
- I thought of using a `BTreeMap` to store the VM versions but I figured we don't really care about the order here

### CODE REVIEW

File: src/jsonrpc/info.rs [hunk 1]
```diff
@@ -369,32 +370,8 @@ impl GetNodeVersionResult {
...
-}
-
-/// ref. <https://docs.avax.network/build/avalanchego-apis/info/#infogetnodeversion>
-#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Clone)]
-#[serde(rename_all = "camelCase")]
-pub struct VmVersions {
-    pub avm: String,
-    pub evm: String,
-    pub platform: String,
-}
```
Comment (richardpringle) [inline src/jsonrpc/info.rs:384]: 🤔  I think we would beneficial to do something like this:
```rust
/// ref. <https://docs.avax.network/build/avalanchego-apis/info/#infogetnodeversion>
#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Clone)]
#[serde(rename_all = "camelCase")]
pub struct VmVersions {
    pub avm: String,
    pub evm: String,
    pub platform: String,
    #[serde(flatten)]
    pub subnet_map: HashMap<String, String>,
}
```
Each key in a `HashMap` is an allocation. If we know that the `avm`, `evm`, and `platform` are always going to be present, then we shouldn't allocate their names. This also simplifies things in terms of accessing `avm`, `evm`, and `platform` and has the added benefit of making the adoption of this breaking change much simpler. 

There are more ways to make this more performant and type-safe (with the above suggestion, you could technically have `String::from("avm")` as a key in the `subnet_map` that would overwrite `thing.avm` during serialization), but I think this should be fine for now.

**Review by richardpringle — CHANGES_REQUESTED** [review]

File: src/jsonrpc/platformvm.rs [hunk 2]
```diff
@@ -768,6 +768,60 @@ impl GetCurrentValidatorsResult {
...
+impl GetPendingValidatorsResponse {
+    pub fn default() -> Self {
+        Self {
+            jsonrpc: "2.0".to_string(),
+            id: 1,
+            result: Some(GetPendingValidatorsResult::default()),
+            error: None,
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (Nuttymoon) [reply src/jsonrpc/platformvm.rs:799]: Perfect. I followed the patterns in place as this is one of the open-source rules :sweat_smile:

File: src/jsonrpc/info.rs (hunk 1 above)
Comment (Nuttymoon) [reply src/jsonrpc/info.rs:384]: Elegant solution :+1: 

I named the HashMap `subnets`

**Review by richardpringle — APPROVED** [review]
LGTM! 

But we should probably also wait for @gyuho's approval

**Pull request merged**